core/src/main/java/com/graphhopper/routing/util/DataFlagEncoder.java

```diff
@@ -296,6 +325,10 @@ public long handleWayTags(ReaderWay way, long allowed, long relationFlags) {
 
             // MAXSPEED
             double maxSpeed = parseSpeed(way.getTag("maxspeed"));
+            if (spatialRuleLookupEnabled() && maxSpeed < 0) {
+                // TODO What if no maxspeed is set, but only forward and backward, and both are higher than the usually allowed?
+                maxSpeed = getSpatialRule(way).getMaxSpeed(way.getTag("highway", ""), maxSpeed);
+            }
             double fwdSpeed = parseSpeed(way.getTag("maxspeed:forward"));
             if (fwdSpeed < 0 || maxSpeed > 0 && maxSpeed < fwdSpeed)
```

Comment:
Is fwdSpeed > maxSpeed an iillegal/unwanted state? From a mapping perspective it shouldn't matter, right?